### PR TITLE
Simple command-line tool to inspect UEFI Signature Database files

### PR DESCRIPTION
**Pull request by out0xb2**
Prints UEFI signature databases in verbose human readable form, or compact readable form for simplified diffing.
Can print the databases as-is, in a canonicalized form, or just the duplicate entries detected in the database.

### CODE REVIEW

**Comment by out0xb2** [comment]
Note:  this change takes a dependency on another PR in pytool-libraries...
https://github.com/tianocore/edk2-pytool-library/pull/118

**Comment by corthon** [comment]
1) What kind of file is this expecting to consume?
2) How does someone produce said file?
3) Can the tool also add certs/fingerprints to a given file (I know it doesn't do that right now, but we have other tools that do and maybe this should be combined)?

**Comment by matthewfcarlson** [comment]
^ Agree with Bret.

It could go into a readme

**Comment by out0xb2** [comment]
Instead of a README, I updated both the docstrings and argparse help.

```
>python sig-db.py dump -h
usage: sig-db.py dump [-h] file

positional arguments:
  file        Filename containing a UEFI Signature Database, a concatenation of EFI_SIGNATURE_LISTs as read from GetVariable([PK, KEK, db, dbx])
```

Generating or manipulating Signature Databases is a feature for another day IMHO.

**Comment by out0xb2** [comment]
Code coverage is failing... am I supposed to add a test for this?  @spbrogan  ?

**Comment by out0xb2** [comment]
@spbrogan , waiting on your review feedback

**Comment by spbrogan** [comment]
Please create a console app so that it can be called from any cli.  https://github.com/tianocore/edk2-pytool-extensions/blob/master/setup.py#L66

**Comment by spbrogan** [comment]
I also think you need a basic feature doc. See info here
https://github.com/tianocore/edk2-pytool-extensions/blob/master/docs/developing.md#new-module-or-class

We might want to create a sub directory in our docs for our CLIs but i'll let you decide if you want to do all that. :)

**Comment by spbrogan** [comment]
sorry one more change.  Need to get pytool-library to release and then up this https://github.com/tianocore/edk2-pytool-extensions/blob/master/setup.py#L78 to match.

**Comment by out0xb2** [comment]
@matthewfcarlson , I will make the final update to bump the library version, _after_ you release a new library.  I'm not in a rush if there are other PRs pending in libraries that you want to get in within a few days...

File: docs/usability/using_sig_db_tool.md
Comment (spbrogan) [inline docs/usability/using_sig_db_tool.md:8]: why does usage not show the positional argument of the actual bin file?  

What is the bin file?  Given how many layers of stuff can be in the Auth variable especially when you want to write it...i think being clear what is the bin file.

Comment (spbrogan) [reply docs/usability/using_sig_db_tool.md:8]: the content from https://github.com/tianocore/edk2-pytool-extensions/pull/215/files#diff-ce00158dd19f56a7155523f2e5ce0fa6R26 would be good

Comment (out0xb2) [reply docs/usability/using_sig_db_tool.md:8]: That is how Python's argparse help works.  It won't tell you the arguments for the subparsers until you request help for each subparser.  If you scroll down in the usage, there is 1 example of each.  However, I will add further examples here in the readme.

Comment (out0xb2) [reply docs/usability/using_sig_db_tool.md:8]: Latest commit below has usage for each action, and a real example for each action

**Comment by out0xb2** [comment]
@matthewfcarlson , @spbrogan , waiting on a pytool library release so that I can bump the version here and complete this PR.  Not urgent at present...

**Comment by matthewfcarlson** [comment]
@out0xb2 working on a pytool library release today. (fingers crossed)

**Comment by out0xb2** [comment]
@matthewfcarlson , I updated the version dependency, and I think that broke the PR gate because even though there is a release of the library, because the release hasn't made it to PyPi yet.  When that happens, I think we should be good to close this.

**Comment by out0xb2** [comment]
@matthewfcarlson , can you kick the PR gates to re-run now the PyPi has published the library?  I don't have that power (unless you want a fake commit).  Thanks!

**Comment by out0xb2** [comment]
@spbrogan, I think we're ready here, unless you want more features

**Comment by out0xb2** [comment]
@matthewfcarlson , pssst, approve it  ;-)

**Review by matthewfcarlson — APPROVED** [review]

**Comment by out0xb2** [comment]
And there was much rejoicing...

**Pull request merged**
